runtime/Java/src/org/antlr/v4/runtime/Vocabulary.java

```diff
@@ -148,4 +157,45 @@
 	 * other user-visible messages which reference specific token types.
 	 */
 	String getDisplayName(int tokenType);
+
+	/** Gets the list of literals ('this' from rules like {@code THIS : 'this';})
+	 *  found in the associated recognizer. Literal at index i is
+	 *  associated with token type i.  Token types without literals
+	 *  (such as {@code ID : [A-Z]+;}) have null entries
+	 *  in the returned array. The first possibly valid entry is
+	 *  {@link Token#MIN_USER_TOKEN_TYPE}.
+	 *
+	 * @return The non-null array of literal names assigned to tokens.
+	 *
+	 * @since 4.6
+	 */
+	String[] getLiteralNames();
```

Comment:
:memo: I'm still 👎 on the change. `Vocabulary` works well in code for representing this information, where `String[]` leads to ambiguity. This change just makes it easier for users to move away from best practices.